main.go

```diff
@@ -78,102 +246,285 @@ func fatal(formatter string, args ...interface{}) {
+		if !filepath.IsAbs(file) {
+			file_errors = append(file_errors, file)
+			log.Printf("error file %s: is not absolute path", file)
+			continue
+		}
+		if plugin.Verbose {
+			log.Printf("Processing: %v", file)
+		}
+		f, err := os.Open(file)
+		if err != nil {
+			file_errors = append(file_errors, file)
+			log.Printf("error couldn't open log file %s: %s", file, err)
+			continue
+		}
+		defer func() {
+			if err := f.Close(); err != nil {
+				file_errors = append(file_errors, file)
+				log.Printf("error couldn't close log file %s: %s", file, err)
+			}
+		}()
+
+		stateFile := filepath.Join(plugin.StateDir, strings.ReplaceAll(file, string(os.PathSeparator), string("_")))
+		if plugin.Verbose {
+			log.Println("stateFile", stateFile)
+		}
+		state, err := getState(stateFile)
+		if err != nil {
+			file_errors = append(file_errors, file)
+			log.Printf("error couldn't get state for log file %s: %s", file, err)
+			continue
+		}
+		// Do we need to reset the state because the requested MatchExpr or InverseMatch is different?
+		resetState := false
+		if state.MatchExpr != "" && state.MatchExpr != plugin.MatchExpr {
+			resetState = true
+		}
+		if state.InverseMatch != plugin.InverseMatch {
+			resetState = true
+		}
+		if resetState {
+			if plugin.EnableStateReset {
+				state = State{}
+				if plugin.Verbose {
+					log.Printf("Info: resetting state file %s because unexpected cached matching condition detected and --reset-state in use", file)
+				}
+			} else {
+				file_errors = append(file_errors, file)
+				log.Printf("Error: state file for %s has unexpected cached matching condition:: Expr: %s Inverse: %v\nEither use --reset-state option, or manually delete state file %s", file, state.MatchExpr, state.InverseMatch, stateFile)
+				continue
+			}
+		}
 		info, err := f.Stat()
 		if err != nil {
-			fatal("%s", err)
+			file_errors = append(file_errors, file)
+			log.Printf("error couldn't get info for file %s: %s", file, err)
+			continue
 		}
-		state.Offset = json.Number(fmt.Sprintf("%d", info.Size()))
-		if err := setState(state, *stateFile); err != nil {
-			fatal("%s", err)
+		// supress alerts on first run (when state file is empty) only when configured (with -ignore-initial-run)
+		if state == (State{}) && plugin.IgnoreInitialRun {
+			state.Offset = json.Number(fmt.Sprintf("%d", info.Size()))
+			state.MatchExpr = plugin.MatchExpr
+			if err := setState(state, stateFile); err != nil {
+				file_errors = append(file_errors, file)
+				log.Printf("error couldn't set state for log file %s: %s", file, err)
+				continue
+			}
+			continue
 		}
-		return
-	}
 
-	offset, _ := state.Offset.Int64()
-	if offset > 0 {
-		if _, err := f.Seek(offset, io.SeekStart); err != nil {
-			fatal("couldn't seek to offset %d: %s", offset, err)
+		offset, _ := state.Offset.Int64()
+		if info.ModTime().Unix() > state.LastTime {
+			if plugin.Verbose {
+				log.Printf("Info: File %s modifed since last read", file)
+			}
+		}
```

Comment:
On systems that have atime enabled, the modification time will be whenever the log file was opened, so I think this will be wrong in those cases.

Luckily, you don't actually need to know the modification time, you only need to know the creation time, and that should reset when the log is rotated. I think you can probably remove LastTime and ModTime and add CreatedTime to the State type without losing any functionality.

Reply:
Let's be clear when you say creation time.. do you mean atime,mtime or ctime and ctime means change time.  
I don't think unix technically defines a create time.  Some filesystems may track creation time, but I don't think its universal.

In fact it looks like golang just exposes a ModTime() in its os agnostic package for dealing with file information. 
https://pkg.go.dev/io/fs#FileInfo

Reply:
removed all logic involving state.LastTime and stateModTIme.